### PR TITLE
ICU-20121 Adding span field for FormattedList.

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20121
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

**Comment by sffc** [comment]
Fixed test failure via `git commit --amend`

File: icu4c/source/i18n/unicode/listformatter.h [hunk 1]
```diff
@@ -65,6 +65,13 @@ struct ListFormatData : public UMemory {
 /**
  * An immutable class containing the result of a list formatting operation.
  *
+ * When calling nextPosition():
+ * The fields are returned from left to right. The special field category
```
Comment (aheninger) [inline icu4c/source/i18n/unicode/listformatter.h]: "left to right" is problematic if the text is right to left. In order from the beginning, or something.

File: icu4c/source/i18n/unicode/ulistformatter.h [hunk 2]
```diff
@@ -111,6 +111,13 @@ ulistfmt_openResult(UErrorCode* ec);
  *
  * You can think of this method as a cast between types.
  *
+ * When calling ufmtval_nextPosition():
+ * The fields are returned from left to right. The special field category
```
Comment (aheninger) [inline icu4c/source/i18n/unicode/ulistformatter.h]: re-word "left to right".

**Comment by sffc** [comment]
Fixed via `git commit --amend`

**Pull request merged**